handler.fish

```diff
@@ -1,24 +1,14 @@
-function __fish_command_not_found_on_interactive --on-event fish_prompt
-    functions --erase __fish_command_not_found_handler
-    functions --erase __fish_command_not_found_setup
-
-    function fish_command_not_found
-        set -l cmd $argv[1]
-        set -l txt (brew which-formula --explain $cmd 2> /dev/null)
-
-        if test -z "$txt"
-            __fish_default_command_not_found_handler $cmd
-        else
-            # https://github.com/fish-shell/fish-shell/issues/159
-            for var in $txt
-                echo $var
-            end
-        end
-    end
-
-    function __fish_command_not_found_handler --on-event fish_command_not_found
-        fish_command_not_found $argv
+function fish_command_not_found
+    set -l cmd $argv[1]
+    set -l txt (brew which-formula --explain $cmd 2> /dev/null)
+
+    if test -z "$txt"
+        __fish_default_command_not_found_handler $cmd
+    else
+        string collect $txt
```

Comment:
`string collect` is a relatively new thing added in [3.1b1](https://fishshell.com/docs/current/relnotes.html#fish-3-1b1-released-january-26-2020) which is close to 1.5 years ago. There could be people still using older versions but it should be rare.